libraries/GCS_MAVLink/MAVLink_routing.cpp

```diff
@@ -130,10 +132,13 @@ bool MAVLink_routing::check_and_forward(GCS_MAVLINK &in_link, const mavlink_mess
         return true;
     }
 
-    if (msg.msgid == MAVLINK_MSG_ID_ADSB_VEHICLE) {
+#if HAL_ADSB_ENABLED
+    // ADSB may be enabled but not compiled in (Rover, Tracker, Blimp), check for presence of the singleton
+    if ((msg.msgid == MAVLINK_MSG_ID_ADSB_VEHICLE) && (AP::ADSB() != nullptr)) {
```

Comment:
I don't think this works - AP::ADSB() will be non-null if compiled in, even if ADSB_TYPE is 0

Reply:
Maybe compiled in was the wrong word, tracker never declares the object, I did check with GDB, it is nullprt on tracker.

Reply:
I have added a enabled check in anycase.